nutanix/resource_nutanix_ndb_clone_refresh.go

```diff
@@ -0,0 +1,114 @@
+package nutanix
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-nutanix/client/era"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+var (
+	EraRefreshCloneTimeout = 15 * time.Minute
+)
+
+func resourceNutanixNDBCloneRefresh() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNutanixNDBCloneRefreshCreate,
+		ReadContext:   resourceNutanixNDBCloneRefreshRead,
+		UpdateContext: resourceNutanixNDBCloneRefreshUpdate,
+		DeleteContext: resourceNutanixNDBCloneRefreshDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(EraRefreshCloneTimeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"clone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"snapshot_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"user_pitr_timestamp"},
+			},
+			"user_pitr_timestamp": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"snapshot_id"},
+			},
+			"timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Asia/Calcutta",
+			},
+		},
+	}
+}
+
+func resourceNutanixNDBCloneRefreshCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	req := &era.CloneRefreshInput{}
+	cloneID := ""
+	if clone, ok := d.GetOk("clone_id"); ok {
+		cloneID = clone.(string)
+	}
+
+	if snapshotID, ok := d.GetOk("snapshot_id"); ok {
+		req.SnapshotID = utils.StringPtr(snapshotID.(string))
+	}
+
+	if userPitrTime, ok := d.GetOk("user_pitr_timestamp"); ok {
+		req.UserPitrTimestamp = utils.StringPtr(userPitrTime.(string))
+	}
+
+	if timezone, ok := d.GetOk("timezone"); ok {
+		req.Timezone = utils.StringPtr(timezone.(string))
+	}
+
+	resp, err := conn.Service.RefreshClone(ctx, req, cloneID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Get Operation ID from response of clone refresh and poll for the operation to get completed.
+	opID := resp.Operationid
+	if opID == "" {
+		return diag.Errorf("error: operation ID is an empty string")
```

Comment:
If there is no operation triggered then do we require this to error out ?

Reply:
Operation is triggered but response doesn't have operation id to poll. so , we throw error saying there is no operation id to poll .